packages/ng/button/button.component.ts

```diff
@@ -37,13 +39,39 @@ export class ButtonComponent implements OnChanges {
 	 */
 	luButton: '' | 'outlined' | 'text' | 'text-invert' = '';
 
+	#iconComponentRef?: ElementRef<HTMLElement>;
+
+	@ContentChild(IconComponent, { read: ElementRef<HTMLElement> })
+	set iconComponentRef(ref: ElementRef<HTMLElement>) {
+		this.#iconComponentRef = ref;
+		this.updateClasses();
+	}
+
+	private get iconOnly(): boolean {
+		const listOfNode = Array.from(this.#elementRef?.nativeElement?.childNodes || []);
+		const noText = listOfNode.every((node) => node.nodeName !== '#text');
```

Comment:
nit, we could use some destructuring
```suggestion
		const noText = listOfNode.every(({ nodeName }) => nodeName !== '#text');
```